src/claviska/SimpleImage.php

```diff
@@ -1079,6 +1079,21 @@ public function text($text, $options, &$boundary = null) {
     return $this;
   }
 
+   /**
+  * Creates Text with Requested Background Image
+  * Passes boundary by reference so boundary from text creation gets updated accordingly when calling text()
+  * Afterwards draws over text() with the correct backgroundColor requested using the rectangle function
+  * Calls text again to draw over the background where the text is supposed to be
+  * Final result shows a text with the backgroundColor requested
+  */
+
+  public function textWithBackgroundColor($text, $options, $backgroundColor, &$boundary = null) {
```

Comment:
I'm not a fan of the naming here, as it's likely too narrow in scope. For example, I can easily see the next request being for something like `textWithBackgroundImage()` and before you know it, we have a bunch of custom text functions that add to the API footprint and make the library harder to work with.

Ideally, this functionality would be baked into `text()` using the `$options` argument:

```php
public function text($text, $options, &$boundary = null) {
   // ...
}
```

This argument is already an array, so an extra property called `backgroundColor` seems to make sense here:

```diff
 $options = array_merge([
   'fontFile' => null,
   'size' => 12,
   'color' => 'black',
+  'backgroundColor' => null,
   'anchor' => 'center',
   'xOffset' => 0,
   'yOffset' => 0,
   'shadow' => null,
   'calculateOffsetFromEdge' => false,
   'baselineAlign' => true
 ], $options);
```

And then the logic you've added here can be tucked inside a conditional that runs only when `$options->backgroundColor`.

It would also be helpful if you wanted to add the argument to [this section of the README](https://github.com/claviska/SimpleImage/blob/master/README.md#texttext-options-boundary), but I can do that if you don't want to.